File: XMLCheck/Handler/CheckRulesXML10/Xml10RuleRunner.cs
using XMLCheck.Handler.CheckRulesXML08;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML10
{
    public static class Xml10RuleRunner
    {
        private static readonly List<IXml10Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(Xml10Model? model,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            foreach (var rule in _rules)
            {
                var err = rule.Check(model,context);
                if (err != null)
                    errors.Add(err);
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml10/xml10RuleRunner.ts
import { Xml10Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml10Rule } from './xml10RuleBase';

export class Xml10RuleRunner {
  private static readonly rules: IXml10Rule[] = [];

  public static run(model: Xml10Model, context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!model) return errors;

    for (const rule of this.rules) {
      const err = rule.check(model, context);
      if (err) {
        errors.push(err);
      }
    }

    return errors;
  }
}
